### ISSUE TITLE
[v1.11.12] gid still not being properly set

### ISSUE THREAD

**Issue by SleepyBrett** [cri-o/cri-o#2219]
**Description**

Using the following manifest:
```
apiVersion: apps/v1beta1
kind: StatefulSet
metadata:
  name: "xxx-w-brett"         # Name of the application. You can append environment (dev, test, prod) to the name too.
  namespace: "default"
  labels:
    app: "dev-w-brett"
    chart: "airflow-1.9.0"
    release: "dev"
spec:
  serviceName: "dev-w-brett"
  podManagementPolicy: "Parallel"
  replicas: 2
  updateStrategy:
    type: RollingUpdate
  template:
    metadata:
      labels:
        app: "dev-w-brett"
        chart: "airflow-1.9.0"
        release: "dev"
      annotations:
        kube2iam.beta.nordstrom.net/role: "arn:aws:iam::542502476764:role/k8s/DS_K8s_Airflow_Access_Role"
    spec: 
      affinity:
        podAntiAffinity:
          preferredDuringSchedulingIgnoredDuringExecution:
          - weight: 10
            podAffinityTerm:
              labelSelector:
                matchExpressions:
                - key: app
                  operator: In
                  values:
                  - dev-w-brett
              topologyKey: beta.nordstrom.net/node-pool
      restartPolicy: Always
      dnsPolicy: Default
      terminationGracePeriodSeconds: 60
      securityContext:
        fsGroup: 10001
        runAsNonRoot: true
        runAsUser: 10001
      containers:
        # AIRFLOW WORKER CONTAINER
        - name: "worker" # Name of container. Can be same as app-name
          image: "quay.io/nordstrom/gidtest:v1"
          imagePullPolicy: "Always"
          securityContext:
            allowPrivilegeEscalation: false
            capabilities:
              drop:
              - ALL
            privileged: false
            runAsNonRoot: true
            runAsUser: 10001
            readOnlyRootFilesystem: true
          args:
          - |
            while true; do sleep 100000; done
          command:
          - bash
          - -c
          env:
            - name: MY_POD_NAME
              valueFrom:
                fieldRef:
                  fieldPath: metadata.name
            - name: MY_POD_NAMESPACE
              valueFrom:
                fieldRef:
                  fieldPath: metadata.namespace
      volumes:
        - name: tmp
          emptyDir: {}
        - name: home
          emptyDir: {}
```
We first noticed this problem in v.1.11.11 but thought one of the patches incorporated into v1.11.12 would solve it. However that is not what we are seeing:

Docker 17.03 -
```
$ id
uid=10001(airflow) gid=10001(airflow) groups=10001(airflow)
```
crio 1.11.11 -
```
$ id
uid=10001(airflow) gid=0(root) groups=0(root),10001(airflow)
```
crio 1.11.12 -
```
$ id
uid=10001(airflow) gid=0(root) groups=0(root),10001(airflow)
```

As you can see we are still seeing the container's user GID set to 0, even though the user is not in that group at all.

```
$ cat /etc/group
root:x:0:
daemon:x:1:
bin:x:2:
...
airflow:x:10001:
nonroot:x:59417:
```

```
$ cat /etc/passwd
root:x:0:0:root:/root:/bin/bash
...
airflow:x:10001:10001:,,,:/home/airflow/:/bin/sh
nonroot:x:59417:59417:,,,:/home/nonroot/:/bin/sh
```

**Steps to reproduce the issue:**
I think it's pretty well covered in the description

**Describe the results you received:**
No change from 1.11.11

**Describe the results you expected:**
```
$ id
uid=10001(airflow) gid=10001(airflow) groups=10001(airflow)
```

**Additional information you deem important (e.g. issue happens only occasionally):**

**Output of `crio --version`:**

```
crio version 1.11.12
commit: "846d94bb0f033ec082d28cc3e786b7469764f08f-dirty"
```

In this case the build is dirty because the following patch was applied to get the build to complete:
```
diff --git a/Dockerfile b/Dockerfile
index 2028b349..1b9ec926 100644
--- a/Dockerfile
+++ b/Dockerfile
@@ -1,7 +1,7 @@
 FROM golang:1.10

 # libseccomp in jessie is not _quite_ new enough -- need backports version
-RUN echo 'deb http://httpredir.debian.org/debian jessie-backports main' > /etc/apt/sources.list.d/backports.list
+RUN echo 'deb [check-valid-until=no] http://archive.debian.org/debian jessie-backports main' > /etc/apt/sources.list.d/backports.list

 RUN apt-get update && apt-get install -y \
     apparmor \
```

**Additional environment details (AWS, VirtualBox, physical, etc.):**
Container Linux by CoreOS 2023.4.0 (Rhyolite) - Kubernetes v1.11.8

**Comment by mrunalp** [comment]
This is a build issue. The fix is actually landed post 1.11.12. We will cut a 1.11.13. Thanks!

**Comment by mrunalp** [comment]
Closing since we cut 1.11.13 with the fix that @SleepyBrett verified as working.
